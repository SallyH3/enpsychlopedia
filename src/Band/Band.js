import React from 'react';

const Band = (props) => {
 return (
   <div className="scroll-container">
    <div className="band-container">
      <img className="band-image" src = {props.bandImg} />
      <h2 className="band-name">{props.bandName}</h2>
    </div>
   </div>
 )
}



export default Band;