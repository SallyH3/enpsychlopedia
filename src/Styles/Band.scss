.band-container {
  width: 260px;
  height: 260px;
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;
}

.band-image {
  padding: 10px;
  width: 100%;
  height: auto;
  opacity: 0.2;
}

.band-name {
  position: relative;
  bottom: 150px;
  padding: 0 10px;
  z-index: 2;
}