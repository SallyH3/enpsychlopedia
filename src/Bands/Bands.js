import React, { Component } from 'react';
import Band from '../Band/Band.js';

export default class Bands extends Component {
  constructor(props) {
    super(props);
    this.state = {
      active: [],
      inactive: []
    }
  }

  assignActiveBands() {
    let activeBands = this.props.genreBands.filter(band => {
        return band.active === true;
      })
    this.setState({active: activeBands})
  }

  assignInactiveBands() {
    let inactiveBands = this.props.genreBands.filter(band => {
        return band.active === false;
      })
    this.setState({inactive: inactiveBands})
  }

  componentWillReceiveProps() {
    this.assignActiveBands();
    this.assignInactiveBands();
  }

  filterAllBands(props) {
    console.log(props)
    return (
      {this.props.genreBands.map((band, index) => (
        <Band key={index}
               bandName={band.name}
               bandImg={band.image}
        />
      ))}
    )
  }

  filterActiveBands() {
    return(
      {this.state.active.map((band, index) => (
        <Band key={index}
                bandName={band.name}
                bandImg={band.image}
            />
          ))}
    )
  }

  filterInactiveBands() {
    return(
      {this.state.inactive.map((band, index) => (
        <Band key={index}
               bandName={band.name}
               bandImg={band.image}
        />
      ))}
    )
  }

  render() { 
  let fallback = this.props.currentSelection;
  if (!fallback) { 
    return (
      <h1>Hi Sally!</h1>
    )
  }
    return (
      <section className='bands-wrapper'>
        if(this.props.currentSelection === 'all') {
          {this.filterAllBands(this.props)}
        } else if (this.props.currentSelection === 'active') {
          {this.filterActiveBands()}
        } else {
          {this.filterInactiveBands()}
        }
      </section>
    )
  }
}