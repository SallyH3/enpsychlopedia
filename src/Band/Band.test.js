import React from 'react';
import Band from './Band';
import Bands from '../Bands/Bands';
import { shallow } from 'enzyme';

let mockBand = {
  name: "Sparklehorse",
  origin: "Richmond Virginia",
  image: "https://i.imgur.com/ciMF4gb.jpg",
  members: [
    "Mark Linkous",
    "Scott Minor"
  ],
  genreId: 1,
  active: false,
  label: "Slow River",
  sideProjects: [
    "Danger Mouse",
    "Sol Seppy",
    "David Lynch",
    "Angela Faye Martin"
  ]
};

const getCurrentBandMock = jest.fn();

describe('Band', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(
      <Band 
        key={1}
        band={mockBand}
        getCurrentBand={getCurrentBandMock}
      />
    )
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should invoke sendBandData when a band is clicked', () => {
    wrapper.find('.band-container').simulate('click', { preventDefault: () => {}});
    expect(getCurrentBandMock).toBeCalled(); 
  });


});