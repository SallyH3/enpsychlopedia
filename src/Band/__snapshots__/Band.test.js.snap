// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Band should match snapshot 1`] = `
<div
  className="scroll-container"
>
  <div
    className="band-container"
    onClick={[Function]}
  >
    <img
      className="band-image"
      src="https://i.imgur.com/ciMF4gb.jpg"
    />
    <h2
      className="band-name"
    >
      Sparklehorse
    </h2>
  </div>
</div>
`;
